docs/ruleset_functions/get_HVAC_building_area_types_and_zones.md

```diff
@@ -0,0 +1,177 @@
+	- `lighting_space_lookup["SPORTS_ARENA_PLAYING_AREA_CLASS_II_FACILITY"] = "PUBLIC_ASSEMBLY"`
+	- `lighting_space_lookup["SPORTS_ARENA_PLAYING_AREA_CLASS_III_FACILITY"] = "PUBLIC_ASSEMBLY"`
+	- `lighting_space_lookup["SPORTS_ARENA_PLAYING_AREA_CLASS_IV_FACILITY"] = "PUBLIC_ASSEMBLY"`
+	- `lighting_space_lookup["TRANSPORTATION_FACILITY_BAGGAGE_CAROUSEL_AREA"] = "OTHER_NON_RESIDENTIAL"`
+	- `lighting_space_lookup["TRANSPORTATION_FACILITY_AIRPORT_CONCOURSE"] = "OTHER_NON_RESIDENTIAL"`
+	- `lighting_space_lookup["TRANSPORTATION_FACILITY_TICKET_COUNTER"] = "OTHER_NON_RESIDENTIAL"`
+	- `lighting_space_lookup["WAREHOUSE_STORAGE_AREA_MEDIUM_TO_BULKY_PALLETIZED_ITEMS"] = "HEATED-ONLY_STORAGE"
+	- `lighting_space_lookup["WAREHOUSE_STORAGE_AREA_SMALLER_HAND_CARRIED_ITEMS"] = "HEATED-ONLY_STORAGE"
+
+- For each building segment in RMR: `for building_segment in RMR.building.building_segments:`
+	- Create list_of_zones for storing zone_ids `list_of_zones = []`
+	- For each zone in building_segment: `for zone in building_segment.zones:`
+		- add zone_id to list_of_zones: `list_of_zones.append(zone.id)`
+		- create space_area_type variable and set equal to "None": `space_area_type = "None"`
+		- check if there is an HVAC Building Area Type: `if building_segment.area_type_heating_ventilation_air_conditioning_system != "Null":`
+			- set space_area_type equal to HVAC BAT: `space_area_type = building_segment.area_type_heating_ventilation_air_conditioning_system`
+		- else:
+			- look for a lighting_building_area_type: `if building_segment.lighting_building_area_type != "Null":`
+				- set space_area_type based on the lookup table: `space_area_type = lighting_space_lookup[building_segment.lighting_building_area_type`
+		- For each space in zone: `for space in zone.spaces:`
+			- check if there is a value in space_area_type: `if space_area_type != "None":`
+				- add space area: `list_buildiung_area_types_with_total_area_and_zones[space_area_type]["AREA"] += space.floor_area`
+			- otherwise, we'll need to map the building area type off of the ligthing category: `else:`
+				- set the space area type based on lighting space area: `space_area_type = lighting_space_lookup[space.lighting_space_type]`
+				- add space to the area based on space.lighting_space_type: `list_buildiung_area_types_with_total_area_and_zones[space_area_type]["AREA"] += space.floor_area`
+
+	- Append zone_ids to the dict: `list_buildiung_area_types_with_total_area_and_zones[space_area_type]["ZONE_IDS"] = list_of_zones`
+	
+	 **Returns** `return list_buildiung_area_types_with_total_area_and_zones`  
+
+**Notes/Questions:**  
+1. "Heated-Only Storage" was not listed in the schema under HeatingVentilatingAirConditioningBuildingAreaOptions2019ASHRAE901 when this function was written - double check spelling
+2. a map was provided for mapping LightingBuildingAreaOptions2019ASHRAE901T951TG38, but not LightingSpaceOptions2019ASHRAE901TG37.  Double check my mapping

```

Comment:
@claperle - can you plese provide mapping for #2 if you haven't already?

Reply:
@mkarpman I provided what we have. We do not have mapping for space types.

Reply:
Yes, I thought about that also. In general, the current schema architecture implies that each BuildingSegment represent one occupancy type . (The main point of this data object is to group zones within a mixed use building by occupancy type to allow specifying  area_type_vertical_fenestration, lighting_building_area_type and area_type_heating_ventilating_air_conditioning_system that apply to all zones within the segment.) Few questions along these lines:

1. For a multifamily building, are we expecting that corridors, stairwells, etc. are included into a separate building segment from apartments?
2. . If area_type_heating_ventilating_air_conditioning_system is provided, should we be checking that it is consistent with the lighting_space_area_type? E.g., if area_type_heating_ventilating_air_conditioning_system is set to residential, but lighting_space_area_type suggests that it cannot be that? Similarly, should we check for general conistence between HVAC, lighting and SWH building area types specified for a given segment? I think we should, otherwise we can overlook impactful mistakes.
3. Should we use area_type_vertical_fenestration and lighting_building_area_type as proxies for a missing area_type_heating_ventilating_air_conditioning_system before we check space type?
4. Since the entire BuildingSegment is meant to represent single building occupancy type, should the function determine a single area_type_heating_ventilating_air_conditioning_system for each BuildingSegment based on the predominant space type?
5. The function returns list_buildiung_area_types_with_total_area_and_zones that are segregated by BuildingSegment. Should the last step of this function be to aggregate the same bluilding area types across segments? E.g., if there are two building segments that are both "Other-Nonresidential", they would be combined for the purpose of determining building area type.

Reply:
> Yes, I thought about that also. In general, the current schema architecture implies that each BuildingSegment represent one occupancy type . (The main point of this data object is to group zones within a mixed use building by occupancy type to allow specifying area_type_vertical_fenestration, lighting_building_area_type and area_type_heating_ventilating_air_conditioning_system that apply to all zones within the segment.) Few questions along these lines:
> 
> 1. For a multifamily building, are we expecting that corridors, stairwells, etc. are included into a separate building segment from apartments?

**I believe that corridors serving multi-family are considered residential HVAC_BAT, right?**

> 2. . If area_type_heating_ventilating_air_conditioning_system is provided, should we be checking that it is consistent with the lighting_space_area_type? E.g., if area_type_heating_ventilating_air_conditioning_system is set to residential, but lighting_space_area_type suggests that it cannot be that? Similarly, should we check for general conistence between HVAC, lighting and SWH building area types specified for a given segment? I think we should, otherwise we can overlook impactful mistakes.

**should this be a separate check or rule?**

> 3. Should we use area_type_vertical_fenestration and lighting_building_area_type as proxies for a missing area_type_heating_ventilating_air_conditioning_system before we check space type?
> 4. Since the entire BuildingSegment is meant to represent single building occupancy type, should the function determine a single area_type_heating_ventilating_air_conditioning_system for each BuildingSegment based on the predominant space type?

**good question.  It is easy to program, just let me know what you would like me to do.**

> 5. The function returns list_buildiung_area_types_with_total_area_and_zones that are segregated by BuildingSegment. Should the last step of this function be to aggregate the same bluilding area types across segments? E.g., if there are two building segments that are both "Other-Nonresidential", they would be combined for the purpose of determining building area type.

**I updated the function for clarity.  It is building-wide**

Reply:
> > 3\. . If area_type_heating_ventilating_air_conditioning_system is provided, should we be checking that it is consistent with the lighting_space_area_type? E.g., if area_type_heating_ventilating_air_conditioning_system is set to residential, but lighting_space_area_type suggests that it cannot be that? Similarly, should we check for general conistence between HVAC, lighting and SWH building area types specified for a given segment? I think we should, otherwise we can overlook impactful mistakes.
> 
> **should this be a separate check or rule?**

Let's discuss with the team. Perhaps this should be a separate rule (and an RDS) as you suggest.

Reply:
> > 1. For a multifamily building, are we expecting that corridors, stairwells, etc. are included into a separate building segment from apartments?
> 
> **I believe that corridors serving multi-family are considered residential HVAC_BAT, right?**

They are actually not considered residential, and are only lumped with residential if they are less than 20K SF.

Reply:
> 3\. Should we use area_type_vertical_fenestration and lighting_building_area_type as proxies for a missing area_type_heating_ventilating_air_conditioning_system before we check space type?

Let's discuss on the call.

Reply:
> 4\. Since the entire BuildingSegment is meant to represent single building occupancy type, should the function determine a single area_type_heating_ventilating_air_conditioning_system for each BuildingSegment based on the predominant space type?

Let's discuss on the call.

Reply:
> I updated the function for clarity. It is building-wide

Can you please point me to where this is done?

Reply:
> There are a few, that are less obvious. For example "CORRIDOR" has sub-categories for hospital and manufacturing, but not for residential or retail, so corridors in these buildings would be categorized as "OTHER-NONRESIDENTIAL" unless we can find a way to classify them better. One way to do this would be to figure out the predominant space type on a given floor (or in a given building) and classify these types of spaces (corridors, copy rooms, storage) as the same BAT as the predominant.

Perhaps this is the way to go. We can tag spaces of type corridor, stairwells, etc. as "supporting nonresidential" and then add them to the predominant non-residentail space type. I am generally leaning toward not using HVAC building area type at all and use lighting space type instead.